pkg/controller/storagecluster/update.go

```diff
@@ -104,6 +124,41 @@ func (c *Controller) rollingUpdate(cluster *corev1.StorageCluster, hash string)
 	return c.syncNodes(cluster, oldPodsToDelete, []string{}, hash)
 }
 
+// function to return the number of unavailable kvdb members and a list of the nodes which should have kvdb running in them
+func (c *Controller) getKVDBNodeAvailability(cluster *corev1.StorageCluster) (int, map[string]bool, error) {
+
+	var kvdbStorageNodeList = make(map[string]bool)
+	unavailableKvdbNodes := 0
+	if storagePodsEnabled(cluster) {
+		storageNodeList, err := c.Driver.GetStorageNodes(cluster)
+		if err != nil {
```

Comment:
As discussed on the call, let's add test cases where GetStorageNodes and GetKVDBMembers returns error.